docs/bazel.md

```diff
@@ -0,0 +1,58 @@
+## Bazel in Pytorch/XLA
+
+[Bazel](https://bazel.build/) is a free software tool used for the automation of building and testing software. [TensorFlow](https://www.tensorflow.org/http) and [OpenXLA](https://github.com/openxla/xla) both use it, which makes it a good fit for PyTorch/XLA as well.
+
+## How to build XLA libraries
+
+Building the libraries is simple:
+
+```bash
+bazel build //third_party/xla_client/...
+```
+
+Bazel is configred via `.bazelrc`, but it can also take flags on the command line.
+
+```bash
+bazel build --config=rbe_cpu_linux_py39 //third_party/xla_client/...
+bazel build --config=rbe_linux_cuda11.8_nvcc_py3.9 //third_party/xla_client/...
+```
+
+The `rbe` configurations build remotely and are cahced and usually faster, but require
+authentication with gcloud. See `.bazelrc` for other `rbe` options.
+
+## Running tests
+
+Bazel is a test plafrom too, making it easy to run tests:
+
+```bash
+bazel test //third_party/xla_client/...
+```
+
+Tests can also be executed remotely in the same fashion as builds.
+
+## Code coverage
+
+When running tests, it can be useful to calculate code coverage.
+
+```bash
+bazel coverage //third_party/xla_client/...
+```
+
+Coverage can be visualized using `lcov` as described in [Bazel's documentation](https://bazel.build/configure/coverage), or in your editor of choice with lcov plugins, e.g. [Coverage Gutters](https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters) for VSCode.
```

Comment:
What are the prereqs for doing this on our source code? Since we already use gtest, would lcov through bazel be seamless?

Reply:
We can already do it for the 3rdparty/xla_client and soon when i migrate the torch_cla/csrc on that as well.I plan to migrate the python tests to bazel as well when that will give way too. Bazel produces lcov, and it can be consed by genhtml or whatever else.